camunda-7-adapter/src/main/java/org/camunda/community/migration/adapter/worker/CamundaPlatform7DelegationWorker.java

```diff
@@ -36,58 +35,57 @@ public void delegateToCamundaPlatformCode(final JobClient client, final Activate
     String delegateExpression = job.getCustomHeaders().get("delegateExpression");
     String expression = job.getCustomHeaders().get("expression");
     String resultVariable = job.getCustomHeaders().get("resultVariable");
-
+    String startListener = job.getCustomHeaders().get("executionListener.start");
+    String endListener = job.getCustomHeaders().get("executionListener.end");
     // and delegate depending on exact way of implementation
-    Map<String, Object> resultPayload = null;
-    DelegateExecution execution = wrapDelegateExecution(job);
-    // this is required as we add the execution to the variables scope for expression evaluation
-    VariableScope variableScope = wrapDelegateExecution(job);
+
+    final DelegateExecution execution = new ZeebeJobDelegateExecution(job);
+
     try {
+      if (startListener == null
+          && endListener == null
+          && delegateClass == null
+          && delegateExpression == null
+          && expression == null) {
+        throw new RuntimeException(
+            "Either 'executionListener.start' or 'class' or 'delegateExpression' or 'expression' or executionListener.end must be specified in task headers for job :"
+                + job);
+      }
+
+      if (startListener != null) {
+        ExecutionListener executionListener =
+            (ExecutionListener) expressionResolver.evaluate(startListener, execution);
+
+        // modifies variables of execution
```

Comment:
I removed the comment completely ... it was indicating that instead of (behavior before this PR) storing changes in a map and using this to complete the job now the variables (if so) are modified inside the delegateExecution.

I think with basic knowledge of what a delegateExpression does and that it has modifiable state, we can just remove any comment here.